sentry_streams/sentry_streams/pipeline/chain.py

```diff
@@ -39,12 +45,22 @@
 )
 from sentry_streams.pipeline.window import MeasurementUnit, Window
 
+logger = logging.getLogger(__name__)
+
+
 TRoute = TypeVar("TRoute")
 
 TIn = TypeVar("TIn")
 TOut = TypeVar("TOut")
 
 
+# a message with a generic payload
+class Message(Generic[TIn]):
```

Comment:
I am trying to intentionally keep `Message` as bare-bones as possible. For example, I don't think the user should have to think about:
- headers after the source step
- partition key
- offset management

However, they can provide a schema for messages and any additional info. e.g. for the generic metrics pipeline, maybe they annotate each message with the metric type. 

Do let me know if something important is missing though.